src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -122,13 +143,13 @@ public PrivilegesEvaluatorResponse evaluate(
                         );
                     }
                     return presponse;
-                } else {
-                    auditLog.logSecurityIndexAttempt(request, action, task);
-                    log.warn("{} for '_all' indices is not allowed for a regular user", action);
-                    presponse.allowed = false;
-                    return presponse.markComplete();
                 }
-            } else if (matchAnySystemIndices(requestedResolved)) {
+                auditLog.logSecurityIndexAttempt(request, action, task);
+                log.warn("{} for '_all' indices is not allowed for a regular user", action);
+                presponse.allowed = false;
+                return presponse.markComplete();
+            }
+            if (matchAnySystemIndices(requestedResolved) && !checkSystemIndexPermissionsForUser(securityRoles)) {
```

Comment:
Seems like this check won't be hit now, the condition is the same as on 122, no?

Reply:
Again a result of conflicts/ reapplying old commits. Thanks for pointing it out